### ISSUE TITLE
pytest-randomly causes occasional CI failures

### ISSUE THREAD

**Issue by josh-hadley** [adobe-type-tools/psautohint#385]
While attempting to [sync dependencies](94a79b9fa8d07988787a69e6394ab47585fe9bcb) between `dev-requirements.txt` and `setup.py`, I re-introduced `pytest-randomly` which was listed in `setup.py` but not in `dev-requirements.txt`. When running the test suite in CI, some test runs failed on the `test_hashmap_old_version` and `test_hashmap_no_version` tests (but were occasionally successful upon running again). I concluded that the random test execution order introduced by `pytest-randomly` was the cause, and removed it. This should be investigated as it suggests that there is some precondition that is getting set when the tests are run in the defined order but which is not always met when run randomly.

**Comment by miguelsousa** [comment]
The test that failed (`test_hashmap_no_version`) checks for the presence of a specific logging message.
By default, psautohint [filters out duplicate logging messages](https://github.com/adobe-type-tools/psautohint/blob/dd01824cc82853c52133bf1abdfa093d078eb8d5/python/psautohint/__main__.py#L390-L407).
To get rid of the gremlins, you may have to implement something similar to [this change](https://github.com/adobe-type-tools/psautohint/pull/347/commits/54d01d3111da832908a1b756ca9f298f9f45b240) throughout the test suite.
A simple way to test this hypothesis would be to disable the dedup filter, and then enable `pytest-randomly`.

Bottom line is: deduping logging messages is desired when running the tools, but causes problems when running the tests.
